src/rgw/rgw_sal_motr.cc

```diff
@@ -2190,99 +2192,138 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
   return rc;
 }
 
-int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+int MotrObject::get_index_key(const DoutPrefixProvider *dpp, std::string& prev_null_obj_key)
 {
   int rc = 0;
+  // Read the null index entry
   string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
   string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
-  int max = 1000;
-  vector<string> keys(max);
-  vector<bufferlist> vals(max);
+  std::string null_ref_key = this->get_name() + "/null";
   bufferlist bl;
   bufferlist::const_iterator iter;
+  rgw_bucket_dir_entry ent_null_ref;
+
+  // Check entry in the cache
+  if (this->store->get_obj_meta_cache()->get(dpp, null_ref_key, bl)) {
+    rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                              M0_IC_GET, null_ref_key, bl);
+    ldpp_dout(dpp, 20) <<__func__<< ": GET null index entry, rc = "<< rc << dendl;
+    if (rc < 0)
+    {
+      ldpp_dout(dpp, 0) << __func__ << "ERROR: Key- "<< null_ref_key <<" does not exist. rc=" << rc << dendl;
+      return rc;
+    }
 
-  if (this->get_bucket()->get_info().versioning_status() == BUCKET_VERSIONED ||
-      this->get_bucket()->get_info().versioning_status() == BUCKET_SUSPENDED) {
+    bufferlist& blr = bl;
+    iter = blr.cbegin();
+    ent_null_ref.decode(iter);
+  }
+  prev_null_obj_key = ent_null_ref.key.name + '[' + ent_null_ref.key.instance + ']';
+  return rc;
+}
 
-    // Check entry in the cache
-    if (this->store->get_obj_meta_cache()->get(dpp, this->get_name(), bl) == 0) {
-        iter = bl.cbegin();
-        ent.decode(iter);
-        rc = 0;
-        goto out;
+int MotrObject::fetch_null_obj_reference(const DoutPrefixProvider *dpp, bufferlist& bl)
```

Comment:
This function is reading object metadata of null object as should be named as `fetch_null_obj` or similar

Reply:
sure, done